lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,248 @@
+        self.data.borrow().fixed_point
+    }
+
+    /// Sets spring's fixed point.
+    pub fn set_fixed_point(&mut self, fixed_point:Vector3<f32>) {
+        self.data.borrow_mut().fixed_point = fixed_point
+    }
+}
+
+// ============================
+// === KinematicProperties ===
+// ============================
+
+struct KinematicData {
+    position     : Vector3<f32>,
+    velocity     : Vector3<f32>,
+    acceleration : Vector3<f32>
+}
+
+/// This structure contains kinematics properties.
+#[derive(Clone)]
+pub struct KinematicProperties {
+    data : Rc<RefCell<KinematicData>>
+}
+
+impl Default for KinematicProperties {
+    fn default() -> Self { Self::new(zero(), zero(), zero()) }
+}
+
+impl KinematicProperties {
+    pub fn new(position:Vector3<f32>, velocity:Vector3<f32>, acceleration:Vector3<f32>) -> Self {
+        let data = Rc::new(RefCell::new(KinematicData { position, velocity, acceleration }));
+        Self { data }
+    }
+
+    /// Gets velocity.
+    pub fn velocity(&self) -> Vector3<f32> {
+        self.data.borrow().velocity
+    }
+
+    /// Sets velocity.
+    pub fn set_velocity(&mut self, velocity:Vector3<f32>) {
+        self.data.borrow_mut().velocity = velocity
+    }
+
+    /// Gets acceleration.
+    pub fn acceleration(&self) -> Vector3<f32> {
+        self.data.borrow().acceleration
+    }
+
+    /// Sets acceleration.
+    pub fn set_acceleration(&mut self, acceleration:Vector3<f32>) {
+        self.data.borrow_mut().acceleration = acceleration
+    }
+}
+
+impl HasPosition for KinematicProperties {
+    fn position(&self) -> Vector3<f32> {
+        self.data.borrow().position
+    }
+
+    fn set_position(&mut self, position:Vector3<f32>) {
+        self.data.borrow_mut().position = position
+    }
+}
+
+// ======================
+// === PropertiesData ===
+// ======================
+
+struct PropertiesData {
+    kinematics : KinematicProperties,
+    spring     : SpringProperties,
+    drag       : DragProperties
+}
+
+impl PropertiesData {
+    pub fn new
+    (kinematics:KinematicProperties, spring:SpringProperties, drag:DragProperties) -> Self {
+        Self { kinematics, spring, drag }
+    }
+}
+
+// ==================
+// === Properties ===
+// ==================
+
+/// A structure including kinematics, drag and spring properties.
+#[derive(Clone)]
+pub struct Properties {
+    data : Rc<RefCell<PropertiesData>>
+}
+
+impl Properties {
+    pub fn new
+    (kinematics:KinematicProperties, spring:SpringProperties, drag:DragProperties) -> Self {
+        let data = Rc::new(RefCell::new(PropertiesData::new(kinematics, spring, drag)));
+        Self { data }
+    }
+
+    pub fn kinematics(&self) -> KinematicProperties { self.data.borrow().kinematics.clone() }
+
+    pub fn set_kinematics(&mut self, kinematics:KinematicProperties) {
+        self.data.borrow_mut().kinematics = kinematics
+    }
+    pub fn spring(&self) -> SpringProperties { self.data.borrow().spring.clone() }
```

Comment:
Let's not use single line functions until you crreate al ot of similar functions and you want to align them to show the similarity. It is arguably hard to read them. I think we got it somewhere in runtime docs.